test_runner/fixtures/neon_fixtures.py

```diff
@@ -1842,6 +1842,23 @@ def assert_no_errors(self):
 
         assert not errors
 
+    def log_contains(self, pattern: str) -> Optional[str]:
```

Comment:
Good point. I believe there is buffering, but I think it's fast enough that it will be very hard to hit that race from python code in practice. I added a comment about it, I think that's enough.